gateway/src/apicast/oauth/oidc.lua

```diff
@@ -30,7 +31,7 @@ local mt = {
 
 local empty = {}
 
-function _M.new(oidc_config)
+function _M.new(oidc_config, service)
```

Comment:
Do we need to pass the `service`? It's only used to obtain the jwt claim with the client ID. Can that attribute be part of the `oidc_config` param?

Reply:
I thought about that, but each service can use different `app_id` so I send the service to have a custom value per service, I think that it's the best way.

Reply:
I see. Even if that's the case, I don't see any reason to couple this class with service when it only needs the name of a jwt claim from it. This constructor already receives `oidc_config`, a table of options, I think that adding another option to that table is a better solution. The caller will need to add that attribute, but I think that's better than introducing a coupling with Service.